core/src/apps/cardano/helpers/hash_builder_collection.py

```diff
@@ -0,0 +1,110 @@
+from apps.common import cbor
+
+if False:
+    from typing import Any, Generic, Generic as GenericKVP, TypeVar
+    from trezor.utils import HashContext
+
+    T = TypeVar("T")
+    K = TypeVar("K")
+    V = TypeVar("V")
+else:
+    T = 0  # type: ignore
+    K = 0  # type: ignore
+    V = 0  # type: ignore
+    Generic = [object]  # type: ignore
```

Comment:
you could do: `Generic = {T: object, (K, V): object}`

Reply:
https://github.com/trezor/trezor-firmware/pull/1683/commits/ed157b3fd72aa0fa789980af3597d201ce21672a